### PR TITLE
NEVISACCESSAPP-4833: Check user enrolment during registration

### PR DESCRIPTION
**Pull request by laszlo-domonkos**

### CODE REVIEW

File: app/src/main/java/ch/nevis/exampleapp/coroutines/common/settings/Settings.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package ch.nevis.exampleapp.coroutines.common.settings
```
Comment (tamas-toth) [inline app/src/main/java/ch/nevis/exampleapp/coroutines/common/settings/Settings.kt:7]: Missing header.

File: app/src/main/java/ch/nevis/exampleapp/coroutines/common/settings/SettingsImpl.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package ch.nevis.exampleapp.coroutines.common.settings
```
Comment (tamas-toth) [inline app/src/main/java/ch/nevis/exampleapp/coroutines/common/settings/SettingsImpl.kt:7]: Missing header.

File: app/src/main/java/ch/nevis/exampleapp/coroutines/data/util/Authenticator.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package ch.nevis.exampleapp.domain.util
```
Comment (tamas-toth) [inline app/src/main/java/ch/nevis/exampleapp/coroutines/data/util/Authenticator.kt]: Missing header.

File: app/src/main/java/ch/nevis/exampleapp/coroutines/ui/selectAuthenticator/model/AuthenticatorItem.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
+package ch.nevis.exampleapp.coroutines.ui.selectAuthenticator.model
```
Comment (tamas-toth) [inline app/src/main/java/ch/nevis/exampleapp/coroutines/ui/selectAuthenticator/model/AuthenticatorItem.kt:7]: Missing header.

File: app/src/main/java/ch/nevis/exampleapp/coroutines/domain/util/Authenticator.kt [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ * Extension function of [Authenticator] to check if it's enrolled for the given user/account or not.
+ *
+ * @param username The username that identifies the user/account.
+ * @param allowClass2Sensors A flag that tells whether using Class 2 biometric sensors is allowed
+ * if the biometric authenticator is selected.
+ * @return A flag that tells whether the authenticator is enrolled for the user or not.
+ * @throws IllegalStateException if the [Authenticator] has an unknown [UserEnrollment] implementation.
+ */
+fun Authenticator.isUserEnrolled(username: String, allowClass2Sensors: Boolean): Boolean {
```
Comment (tamas-toth) [inline app/src/main/java/ch/nevis/exampleapp/coroutines/domain/util/Authenticator.kt:21]: This is duplicated, exactly the same extension can be found in the `ch.nevis.exampleapp.domain.util` package. Do we need both of them?

File: app/src/main/java/ch/nevis/exampleapp/coroutines/common/settings/Settings.kt (hunk 1 above)
Comment (laszlo-domonkos) [reply app/src/main/java/ch/nevis/exampleapp/coroutines/common/settings/Settings.kt:7]: Fixed.

File: app/src/main/java/ch/nevis/exampleapp/coroutines/common/settings/SettingsImpl.kt (hunk 2 above)
Comment (laszlo-domonkos) [reply app/src/main/java/ch/nevis/exampleapp/coroutines/common/settings/SettingsImpl.kt:7]: Fixed.

File: app/src/main/java/ch/nevis/exampleapp/coroutines/domain/util/Authenticator.kt (hunk 5 above)
Comment (laszlo-domonkos) [reply app/src/main/java/ch/nevis/exampleapp/coroutines/domain/util/Authenticator.kt:21]: Thanks for spotting this. I've deleted the one from `ch.nevis.exampleapp.data.util`. It was accidentally copied to there.

File: app/src/main/java/ch/nevis/exampleapp/coroutines/ui/selectAuthenticator/model/AuthenticatorItem.kt (hunk 4 above)
Comment (laszlo-domonkos) [reply app/src/main/java/ch/nevis/exampleapp/coroutines/ui/selectAuthenticator/model/AuthenticatorItem.kt:7]: Fixed.

**Review by tamas-toth — APPROVED** [review]

File: app/src/main/java/ch/nevis/exampleapp/coroutines/domain/interaction/AuthenticationAuthenticatorSelectorImpl.kt [hunk 6]
```diff
@@ -41,17 +50,38 @@ class AuthenticationAuthenticatorSelectorImpl(
...
     }
     //endregion
+
+    //region Private Interface
+    private fun mapForAuthentication(
+        authenticator: Authenticator,
+        context: AuthenticatorSelectionContext
+    ): AuthenticatorItem? {
+        return if (authenticator.registration()
+                .isRegistered(context.account().username()) && authenticator.isSupportedByHardware
```
Comment (rolandh15) [inline app/src/main/java/ch/nevis/exampleapp/coroutines/domain/interaction/AuthenticationAuthenticatorSelectorImpl.kt:73]: Would be good to have a log statement here, this way, the devs can pinpoint where the check happens.

**Review by rolandh15 — APPROVED** [review]

File: app/src/main/java/ch/nevis/exampleapp/coroutines/domain/interaction/RegistrationAuthenticatorSelectorImpl.kt [hunk 7]
```diff
@@ -71,25 +79,38 @@ class RegistrationAuthenticatorSelectorImpl(
 
         val canRegisterBiometric = authenticators.any {
```
Comment (rolandh15) [inline app/src/main/java/ch/nevis/exampleapp/coroutines/domain/interaction/RegistrationAuthenticatorSelectorImpl.kt:81]: I would add the log messages we talked about here too.

Comment (laszlo-domonkos) [reply app/src/main/java/ch/nevis/exampleapp/coroutines/domain/interaction/RegistrationAuthenticatorSelectorImpl.kt:81]: Fixed. I also added it to simplified example app. https://github.com/nevissecurity/nevis-mobile-authentication-sdk-example-app-android/pull/5

File: app/src/main/java/ch/nevis/exampleapp/coroutines/domain/interaction/AuthenticationAuthenticatorSelectorImpl.kt (hunk 6 above)
Comment (laszlo-domonkos) [reply app/src/main/java/ch/nevis/exampleapp/coroutines/domain/interaction/AuthenticationAuthenticatorSelectorImpl.kt:73]: Fixed. I also added it to simplified example app. https://github.com/nevissecurity/nevis-mobile-authentication-sdk-example-app-android/pull/5

**Pull request merged**